pkg/rhoda/rhoda_test.go

```diff
@@ -18,165 +22,379 @@ import (
 
 	Context("Check operator installation", func() {
-		It("dbaasplatforms.dbaas.redhat.com CRD exists", func() {
-			//running it locally, assuming we are outside the cluster
-			var err error
-			if os.Getenv("KUBERNETES_SERVICE_HOST") == "" {
-				var kubeconfig *string
-				if home := homedir.HomeDir(); home != "" {
-					kubeconfig = flag.String("kconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
-				} else {
-					kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
-				}
-				flag.Parse()
-
-				// use the current context in kubeconfig
-				config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
-			} else {
-				config, err = rest.InClusterConfig()
-			}
-			Expect(err).NotTo(HaveOccurred())
-
-			apiextensions, err := apiserver.NewForConfig(config)
-			Expect(err).NotTo(HaveOccurred())
-
+		apiextensions, err := apiserver.NewForConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+		It("Should pass when operator installation is validated", func() {
+			fmt.Println("checking operator installation")
 			// Make sure the CRD exists
 			_, err = apiextensions.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), "dbaasplatforms.dbaas.redhat.com", meta.GetOptions{})
-
+			if err != nil {
+				AbortSuite(err.Error())
+			}
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
 
-	Context("Populate providers array", func() {
-		It("Get list of providers from the vault, create secrets, populate array", func() {
-			//Get ci-secret's data
-			clientset, err := kubernetes.NewForConfig(config)
-			Expect(err).NotTo(HaveOccurred())
-
-			ciSecret, err := clientset.CoreV1().Secrets("osde2e-ci-secrets").Get(context.TODO(), "ci-secrets", meta.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
+	Context("Get all the providers and loop through it to create Secrets and Inventory", func() {
+		var providerNames []string
+		var providers []rhoda.ProviderAccount
+		//Get ci-secret's data
+		clientset, err := kubernetes.NewForConfig(config)
+		Expect(err).NotTo(HaveOccurred())
+		ciSecret, err := clientset.CoreV1().Secrets("osde2e-ci-secrets").Get(context.TODO(), "ci-secrets", meta.GetOptions{})
+		if err != nil {
+			AbortSuite(err.Error())
+		}
+		fmt.Println("ciSecret Found: ")
 
-			fmt.Println("ciSecret Found: ")
-			//get the list of providers by getting providerList secret
-			var providerNames []string
-			if providerListSecret, ok := ciSecret.Data["providerList"]; ok {
-				fmt.Printf("providerListSecret = %s, ok = %v\n", providerListSecret, ok)
-				providerNames = strings.Split(string(providerListSecret), ",")
-				fmt.Println(providerNames)
+		//get the list of providers by getting providerList secret
+		if providerListSecret, ok := ciSecret.Data["providerList"]; ok {
+			//fmt.Printf("providerListSecret = %s, ok = %v\n", providerListSecret, ok)
+			providerNames = strings.Split(string(providerListSecret), ",")
+			fmt.Println(providerNames)
```

Comment:
I think it would be good to create a function that takes providerNames and ciSecret, and returns providers. Move the next block into that function, and have a one liner here to get providers from it. Then, providerNames becomes a local variable to this if block